lexer/lex_test.go

```diff
@@ -760,6 +853,108 @@ func TestIllegalMemberAccess(t *testing.T) {
 	})
 }
 
+func TestInvalidStrings(t *testing.T) {
+
+	cases := []TestCase{
+		{
+			name: "SingleDoubleQuote",
+			code: Str(`"`),
+			want: []lexer.Tokval{illegalToken(`"`)},
+		},
+		{
+			name: "NoEndingDoubleQuote",
+			code: Str(`"dsadasdsa123456`),
+			want: []lexer.Tokval{illegalToken(`"dsadasdsa123456`)},
+		},
+	}
+
+	for _, lineTerminator := range lineTerminators() {
+		code := fmt.Sprintf(`"head%stail"`, lineTerminator.val)
+		cases = append(cases, TestCase{
+			code: Str(code),
+			name: "NewlineTerminator" + lineTerminator.name,
+			want: []lexer.Tokval{illegalToken(code)},
+		})
+	}
+
+	runTests(t, cases)
+}
+
+func TestInvalidSemiColon(t *testing.T) {
+
+	t.Skip("TODO")
+
+	cases := []TestCase{
+		{
+			name: "AfterLeftParen",
+			code: Str("(;"),
+			want: tokensErr(t, leftParenToken(), ";"),
```

Comment:
@tiago4orion what do you think about the idea of modeling a tokensErr like this ? It would avoid to all "illegalToken" all the time, but I the function interface is pretty odd (interface{} where the values are tokens except the last one that is a string).